### ISSUE TITLE
Exclude Management Group Subscription Association

### ISSUE THREAD

**Issue by Eslam10** [Azure/terraform-azurerm-caf-enterprise-scale#209]
### Community Note
- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions
**terraform**: 1.0.9

**azure provider**: 2.82.0

**module**: 1.0.0

### Description
We have a different TF code/pipeline to provision new Azure subscriptions, we would like to assign the subscription to different management groups on the same TF code. Since we keep `subscription_ids` on this azure-caf-es module empty, the module remove the subscriptions from the management group every time it runs.
Is there a way to add `life_cycle, ignore_changes` to this management group subscription association of this module ? 

**Steps to Reproduce**
1. Associate subscriptions to management group outside of caf-es module.
2. Trigger caf-es module.
3. Subscription will be removed from the associated management group in step 1.

**Comment by krowlandson** [comment]
Unfortunately not, as Subscription placement is a part of the governance model.

Have you considered having the output of your Azure subscription pipeline (or state file) as an input to the pipeline deploying this module so you can read in the values and pass them to the expected Management group?

**Comment by Eslam10** [comment]
Hi @krowlandson ,
Thanks, I will try this then, maybe in this case I'll still need to pass the expected management group as an input to the azure subscription pipeline as well.

**Comment by krowlandson** [comment]
That sounds like a good idea.

As long as the subscription pipeline provides the `caf-enterprise-scale` pipeline with the mapping from the point of creation onwards, I see no reason why you cannot also have the Subscription placed directly into the target Management Group.

We are looking to publish some guidance on this at some point in the future, so any feedback you can provide around this scenario would be greatly appreciated.

**Comment by Eslam10** [comment]
Yes. So below is what I have tested so far, which may help in the guidance and help anyone needs the same setup,

- Separate code to provision new Azure EA subscription using `azurerm_subscription` resource.
- The input variable will be a list of subscriptions with management group name of each subscription and some other variables that we need to define as well, example below
````hcl
{
  "subscription1" = {
    "management_group" = "internal-nonprod"
    "region"                       = "westeurope"
    ....
  },
  "subscription2" = {
    "management_group" = "connectivity"
    "region"                       = "westeurope"
    ....
  }
}
````
- In the outputs.tf file, output list of subscription ids per management group, example below,
````hcl
output "subscriptions_internal_nonprod" {
  value = [
    for subscription_name, sub_details in local.subscriptions : azurerm_subscription.main[subscription_name].subscription_id
    if sub_details.management_group == "internal-nonprod"
  ]
}
````

- I am using terraform cloud pipelines, so in the terraform code that uses ES-CAF module, I used data source `terraform_remote_state` to get the output from the subscriptions provisioning pipeline. 

- In ES-CAF module, I am using a `custom_landing_zones`, I defined a local variable for subscription ids, which is getting list of subscription ids for each MG, for example,
````hcl
"internal-nonprod"  =  data.terraform_remote_state.main.outputs.subscriptions_internal_nonprod
```` 
- Then I added this to the `custom_landing_zones.subscription_ids` variable.

This achieved not manually adding the newly created subscription to the ES-CAF TF module.
